cli/azd/pkg/config/config_test.go

```diff
@@ -10,39 +10,104 @@ import (
 func Test_SaveAndLoadConfig(t *testing.T) {
 	defer deleteExistingConfig()
 
-	newConfig := &Config{
-		Account: &Account{
-			DefaultSubscription: &Subscription{
-				Id:   "SUBSCRIPTION_ID",
-				Name: "Test Subscription",
-			},
-			DefaultLocation: &Location{
-				Name:        "eastus2",
-				DisplayName: "East US 2",
+	var azdConfig Config = NewConfig(
+		map[string]any{
+			"defaults": map[string]any{
+				"location":     "eastus2",
+				"subscription": "SUBSCRIPTION_ID",
 			},
 		},
-	}
+	)
 
-	err := newConfig.Save()
+	err := azdConfig.Save()
 	require.NoError(t, err)
 
 	existingConfig, err := Load()
 	require.NoError(t, err)
 	require.NotNil(t, existingConfig)
-	require.Equal(t, newConfig, existingConfig)
+	require.Equal(t, azdConfig, existingConfig)
 }
 
 func Test_SaveAndLoadEmptyConfig(t *testing.T) {
 	defer deleteExistingConfig()
 
-	newConfig := &Config{}
-	err := newConfig.Save()
+	azdConfig := NewConfig(nil)
+	err := azdConfig.Save()
 	require.NoError(t, err)
 
 	existingConfig, err := Load()
 	require.NoError(t, err)
 	require.NotNil(t, existingConfig)
-	require.Nil(t, existingConfig.Account)
+}
+
+func Test_SetGetUnsetWithValue(t *testing.T) {
+	tests := []struct {
+		name  string
+		path  string
+		value any
+	}{
+		{
+			name:  "RootValue",
+			path:  "a",
+			value: "apple",
+		},
+		{
+			name:  "NestedValue",
+			path:  "defaults.location",
+			value: "westus",
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			azdConfig := NewConfig(nil)
+			err := azdConfig.Set(test.path, test.value)
+			require.NoError(t, err)
+
+			value, ok := azdConfig.Get(test.path)
+			require.True(t, ok)
+			require.Equal(t, test.value, value)
+
+			err = azdConfig.Unset(test.path)
+			require.NoError(t, err)
+
+			value, ok = azdConfig.Get(test.path)
+			require.Nil(t, value)
+			require.False(t, ok)
+		})
+	}
+}
+
+func Test_SetGetUnsetRootNodeWithChildren(t *testing.T) {
+	expectedLocation := "westus2"
+	expectedEmail := "john.doe@contoso.com"
+
+	azdConfig := NewConfig(nil)
+	_ = azdConfig.Set("defaults.location", expectedLocation)
+	_ = azdConfig.Set("defaults.subscription", "SUBSCRIPTION_ID")
```

Comment:
What's the discoverability experience for the user here?  How can customer know what keys are actually valid `config` keys?
How do we prevent typos like `azd config set defults.location home` ??

Can we improve the experience here by adding commands and help for each supported configuration??  (so the customer doesn't have to learn about the config from documentation).

And we can go from `azd config set <key> <value>` to `azd config defaults location <value>`???

Then, customers don't have to learn keys in advance (or set invalid keys).  They could do `azd config defaults --help` to see what you can do.

Yes, it's more work for developers, but isn't it a better experience for customers?

Reply:
This is meant to be a general purpose configuration command set without any custom commands for specific keys.  `azd` will internally set the correct keys for our defaults and user can refer to documentation for list of supported keys.

Reply:
This is one reason why we chose to not develop/copy the `az account` command category.  We can always add more commands in the future if we find there is a need.